core/trino-main/src/test/java/io/trino/sql/gen/TestExpressionCompiler.java

```diff
@@ -1476,6 +1483,27 @@ public void testFunctionCallRegexp()
         Futures.allAsList(futures).get();
     }
 
+    @Test
+    public void testSearchInterruptible()
+            throws Exception
+    {
+        String pattern = "\\((.*,)+(.*\\))";
+        String source = Files.readString(Paths.get(EXTRA_LONG_SOURCE_PATH));
+        RegularExpressionRunnable regularExpressionRunnable = new RegularExpressionRunnable(source, pattern);
+        Thread searchChildThread = new Thread(regularExpressionRunnable);
+
+        searchChildThread.start();
+
+        // wait for the child thread to make some progress
+        searchChildThread.join(1000);
+        searchChildThread.interrupt();
+
+        // wait for child thread to get in to terminated state
+        searchChildThread.join();
+        Assert.assertEquals(regularExpressionRunnable.exceptionClause.getErrorCode().getType(), ErrorType.USER_ERROR);
```

Comment:
interesting, visibility of `exceptionClause` update is always guaranteed here after interruption, so this won't be flaky: https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.4.4

> If thread T1 interrupts thread T2, the interrupt by T1 synchronizes-with any point where any other thread (including T2) determines that T2 has been interrupted (by having an InterruptedException thrown or by invoking Thread.interrupted or Thread.isInterrupted).

Reply:
nit: static import for assertequals